### DISCUSSION TITLE
Auth/user info plugin for NetAuth

### DISCUSSION THREAD

**Discussion by classabbyamp** [buildbot/buildbot#8030] [Show and tell]
Void Linux uses [NetAuth](https://netauth.org) for SSO, so I made a [buildbot plugin](https://pypi.org/project/buildbot-netauth/) for our buildbot instance.

![image](https://github.com/user-attachments/assets/2ca6ebb5-ab64-489e-bf45-b219244e4a48)

**Comment by p12tic** [comment]
@classabbyamp Would you like to submit this plugin to Buildbot proper?

**Reply by classabbyamp** [reply]
I'll consider it. There are a few improvements I'd maybe like to make, and I'd like to ensure it works well on our buildbot for a bit first.

**Reply by p12tic** [reply]
Thanks. The reason why I'm asking is that code distribution with upstream project is much better in the perspective of end users most of the time. There is no need to check Buildbot version compatibilities, there's no need to verify that the project and the maintainer won't attempt to distribute malware, there is no need to second guess chances that the project will become unmaintained.

**Reply by classabbyamp** [reply]
I understand. This was mostly developed to use within our own deployment so those concerns don't necessarily apply.

As an aside, is it possible to implement this with a proper html login form instead of a basic auth prompt? It would improve the UX. (if not, would you accept changes to implement that?)
